bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java

```diff
@@ -0,0 +1,358 @@
+                logger.warn("A non numeric hue ID '{}' was assigned. Ignoring!", metadata.getValue());
+            }
+        }
+    }
+
+    /**
+     * Although hue IDs are strings, a lot of implementations out there assume them to be numbers. Therefore
+     * we map each item to a number and store that in the meta data provider.
+     *
+     * @param item The item to map
+     * @return A stringified integer number
+     */
+    public String mapItemUIDtoHueID(Item item) {
+        MetadataKey key = new MetadataKey(METAKEY, item.getUID());
+        Metadata metadata = metadataRegistry.get(key);
+        int hueId = 0;
+        if (metadata != null) {
+            try {
+                hueId = Integer.parseInt(metadata.getValue());
+            } catch (NumberFormatException e) {
+                logger.warn("A non numeric hue ID '{}' was assigned. Ignore and reassign a different id now!",
+                        metadata.getValue());
+            }
+        }
+        if (hueId == 0) {
+            ++highestAssignedHueID;
+            hueId = highestAssignedHueID;
+            metadataRegistry.add(new Metadata(key, String.valueOf(hueId), null));
+        }
+
+        return String.valueOf(hueId);
+    }
+
+    public boolean isReady() {
+        return discoveryIps.size() > 0;
+    }
+
+    public HueEmulationConfig getConfig() {
+        return config;
+    }
+
+    public int getHighestAssignedHueID() {
+        return highestAssignedHueID;
+    }
+
+    /**
+     * Sets the link button state.
+     *
+     * Starts a pairing timeout thread if set to true.
+     * Stops any already running timers.
+     *
+     * @param linkbutton New link button state
+     */
+    public void setLinkbutton(boolean linkbutton, boolean createUsersOnEveryEndpoint,
+            boolean temporarilyEmulateV1bridge) {
+        ds.config.linkbutton = linkbutton;
+        config.createNewUserOnEveryEndpoint = createUsersOnEveryEndpoint;
+        if (temporarilyEmulateV1bridge) {
+            ds.config.makeV1bridge();
+        } else if (!config.permanentV1bridge) {
+            ds.config.makeV2bridge();
+        }
+        ScheduledFuture<?> future = pairingOffFuture;
+        if (future != null) {
+            future.cancel(false);
+        }
+        if (!linkbutton) {
+            logger.info("Hue Emulation pairing disabled");
+            return;
+        }
+
+        logger.info("Hue Emulation pairing enabled for {}s", ds.config.networkopenduration);
+        pairingOffFuture = scheduler.schedule(() -> {
+            logger.info("Hue Emulation disable pairing...");
+            if (!config.permanentV1bridge) { // Restore bridge version
+                ds.config.makeV2bridge();
+            }
+            config.createNewUserOnEveryEndpoint = false;
+            config.temporarilyEmulateV1bridge = false;
+            WriteConfig.unsetPairingMode(configAdmin);
+        }, ds.config.networkopenduration * 1000, TimeUnit.MILLISECONDS);
```

Comment:
I have seen that in a lot of binding. Why not use `TimeUnit.SECONDS` instead?

Reply:
Just used to milliseconds. Before we had TimeUnit APIs there were only millisecond based APIs available.